test/routs/testSoldierRouts.js

```diff
@@ -0,0 +1,94 @@
+const expect = require('chai').expect;
+const assert = require('assert');
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const testServer = require("../../start.js").server;
+const MongoClient = require('mongodb').MongoClient;
+const DBurl = 'mongodb://localhost:27017';
+const dbName = 'myproject';
+
+describe("Soldiers tests", function () {
+  const serverUrl = 'http://localhost:3000/soldiers';
+
+  after("Closing the server", function (doneAfter) {
+    testServer.close();
+    doneAfter();
+  });
+
+  describe("Soldiers post test", function () {
+
+    it("Should be able to return correct respone when the soldier data is missing fields", function (done) {
+      const Http = new XMLHttpRequest();
+      const url = serverUrl;
+      Http.open("POST", url);
+      Http.send(JSON.stringify({
+        "id": "tt8145643",
+        "name": "lior",
+        "limitations": ["yeshiva"]
+      }));
+      Http.onreadystatechange = (err) => {
+        if (Http.readyState == 4 && Http.status == 200) {
```

Comment:
If the status code isn't 200 you never call done and you'll get a timeout error instead of an informative one